service/src/main/java/org/broadinstitute/listener/relay/inspectors/RequestLogger.java

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.listener.relay.inspectors;
+
+import com.microsoft.azure.relay.RelayedHttpListenerRequest;
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.TreeMap;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.listener.relay.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RequestLogger {
+
+  private final Logger logger = LoggerFactory.getLogger(RequestLogger.class);
+  private static final List<String> MUST_MASKED_HEADER_NAMES = List.of("Authorization", "Cookie");
+
+  /**
+   * Logs a relayed HTTP request with the result status code
+   *
+   * @param relayedHttpListenerRequest Relayed request
+   * @param statusCode Result of the request
+   * @param requestTimestamp Timestamp of the request
+   * @param prefix Logging prefix to include
+   * @throws IOException
+   * @throws InterruptedException
+   */
+  public void logRequest(
+      RelayedHttpListenerRequest relayedHttpListenerRequest,
+      int statusCode,
+      OffsetDateTime requestTimestamp,
+      String prefix)
+      throws IOException, InterruptedException {
+    if (relayedHttpListenerRequest == null) {
+      logger.warn("Null request provided for logging");
+      return;
+    }
+
+    // HTTP headers are case-insensitive
+    var headers = new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER);
+    headers.putAll(relayedHttpListenerRequest.getHeaders());
+
+    var referer = headers.getOrDefault("Referer", "-");
+    var origin = headers.getOrDefault("Origin", "-");
+    var ua = headers.getOrDefault("User-Agent", "-");
+
+    String endpoint = "";
+    if (relayedHttpListenerRequest.getRemoteEndPoint() != null) {
+      endpoint = relayedHttpListenerRequest.getRemoteEndPoint().getHostString();
+    }
+    var claims = getTokenClaims(headers);
+
+    // log in a single apache-ish line
+    logger.info(
+        "{} {} {} \"{}\" [{}] {} \"{} {}\" {} \"{}\" \"{}\" \"{}\"",
```

Comment:
```suggestion
        "{} - {} {} \"{}\" \"-\" [{}] {} {} \"{} {} {}\" {} {} \"{}\" \"{}\" \"{}\"",
```
Based on https://github.com/broadinstitute/terra-helmfile/blob/d0d482c72376174d35b773e93e5d54786d0c9dd8/charts/httpd-proxy/templates/_proxy_configmap.yaml#LL18C5-L18C192 
>
```apache
LogFormat "%{X-Forwarded-For}i %l %u \"%{OIDC_CLAIM_email}i\" \"%{X-App-ID}i\" [%{%FT%T}t.%{msec_frac}t%{%z}t] %D %X \"%r\" %>s %b \"%{Referer}i\" \"%{Origin}i\" \"%{User-Agent}i\"" proxy
```
Would be good to plug these:
> %l Remote logname (from identd, if supplied). This will return a dash unless [mod_ident](https://httpd.apache.org/docs/2.4/mod/mod_ident.html) is present and [IdentityCheck](https://httpd.apache.org/docs/2.4/mod/mod_ident.html#identitycheck) is set On.
>
> %D The time taken to serve the request, in microseconds.
>
> %X | Connection status when response is completed:
> X = | Connection aborted before the response completed.
> + = | Connection may be kept alive after the response is sent.
> - = | Connection will be closed after the response is sent.
>
> %r First line of request.
>
I think for `%r` we also log which HTTP version it is, like `GET /status HTTP/1.1`, would be helpful to add that here.

> %b | Size of response in bytes, excluding HTTP headers. In CLF format, i.e. a '-' rather than a 0 when no bytes are sent.